reframe/utility/typecheck.py

```diff
@@ -342,6 +342,35 @@ def __rfm_cast_str__(cls, s):
         return s
 
 
+class Bool(metaclass=_BuiltinType):
+    _type = bool
```

Comment:
Restrict further the type:

```suggestion
    _type = bool
    __slots__ = ('__val',)
```